python/cudf_polars/tests/test_string.py

```diff
@@ -59,3 +60,43 @@ def test_contains_invalid(ldf, pat):
         query.collect()
     with pytest.raises(pl.exceptions.ComputeError):
         query.collect(post_opt_callback=partial(execute_with_cudf, raise_on_fail=True))
+
+
+def test_replace_literal(ldf):
+    query = ldf.select(pl.col("a").str.replace("A", "a", literal=True))
+    assert_gpu_result_equal(query)
+
+
+def test_replace_re(ldf):
+    query = ldf.select(pl.col("a").str.replace("A", "a", literal=False))
+    with pytest.raises(NotImplementedError):
+        _ = translate_ir(query._ldf.visit())
+
+
+@pytest.mark.parametrize(
+    "target,repl",
+    [
+        (["A", "de", "kLm", "awef"], "a"),
+        (["A", "de", "kLm", "awef"], ["a", "b", "c", "d"]),
+        (pl.col("a"), pl.col("a")),
+    ],
+)
+def test_replace_many(ldf, target, repl):
+    if isinstance(target, pl.Expr):
+        # libcudf cannot handle nulls in target column
+        # TODO: refactor so that drop_nulls happens on the pl.col call
+        # (not possible right now since the dropnull Expr function is not
+        # implemented)
+        ldf = ldf.drop_nulls()
```

Comment:
What are the semantics from polars that we want to match in the case that the target or replacement columns have nulls?